tests/library/test_json.py

```diff
@@ -363,6 +363,140 @@ def test_bad_object(self, bad_string, failure_byte):
         _check_match_failure(bad_string, failure_byte, schema_obj)
 
 
+class TestArrayWithLengthConstraints:
+    prefix_schema_obj = [
+        {"type": "integer"},
+        {"type": "boolean"}
+    ]
+    items_schema_obj = {"type": "string"}
+
+    @pytest.mark.parametrize(
+        "min_items, max_items, target_obj",
+        [
+            (0, 0, []), # None allowed, none provided
+            (0, 1, []), # Some prefixItems allowed, none provided
+            (0, 1, [42]), # Some prefixItems allowed, one provided
+            (1, 2, [42, True]),  # All prefix items, no extra items allowed or provided.
+            (1, 3, [42, True]),  # All prefix items, some extra items allowed but not provided.
+            (0, 3, [42, True, "hello"]),  # All prefix items and one extra item
+            (3, 4, [42, True, "hello"]),  # All prefix items and one extra item but more allowed
+            (5, 5, [42, True, "hello", "world", "test"]),  # Exactly meets minItems and maxItems.
+            (0, 10, [42, True] + ["extra"] * 8),  # Exactly meet large number of extra items
+        ]
+    )
+    def test_good_with_prefix_and_items(self, min_items, max_items, target_obj):
+        schema_obj = {
+            "prefixItems": self.prefix_schema_obj,
+            "items": self.items_schema_obj,
+            "minItems": min_items,
+            "maxItems": max_items,
+            "type": "array",
+        }
+        _generate_and_check(target_obj, schema_obj)
```

Comment:
Given the prefixItems, is there an extra check you can make if `_generate_and_check()` is adjusted to return the generated list?

Reply:
Can you clarify what you mean? Just whether "returning more with the error" would let us check more with the test?

Reply:
Sorry, yes. I was overthinking this, and forgetting that the 'generation' is fully deterministic.

Reply:
Ah, no worries